### DISCUSSION TITLE
Directus deploy in digitalocean

### DISCUSSION THREAD

**Discussion by nehadnb** [directus/directus#16839] [Archive]
### Describe the Bug

I have repo with packege.json, package-lock.json and .gitignore file

```
{
  "name": "directus",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "directus start"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "directus": "^9.21.2"
  },
  "engines": {
    "node": "^16.0.0",
    "npm": "^8.5.0"
  }
}
```
Also environment variable have all db details like DB_HOST, DB_PORT, DB_DATABASE, DB_USER,DB_PASSWORD
![image](https://user-images.githubusercontent.com/109754993/208171388-d753255f-215d-4e3d-aec9-f47a4adf8962.png)

![image](https://user-images.githubusercontent.com/109754993/208171529-3f6ecabc-9ea9-4585-8268-ae0d67fcb10f.png)
throw above error while deploying 

If i remove npx directus bootstrap from build command then it gives me this error

![image](https://user-images.githubusercontent.com/109754993/208194049-43016e15-bd78-4e40-b04e-bcca52298e5c.png)

Can any one let me know what am i missing for directus setup in digitalocean

### To Reproduce

Follow steps - https://docs.directus.io/self-hosted/installation/digitalocean-app-platform.html
add package.json same as me 

### What version of Directus are you using?

9.21.2

### What version of Node.js are you using?

16.18.1

### What database are you using?

 Postgres 14

### What browser are you using?

Chrome

### How are you deploying Directus?

Digital Ocean App Platform

**Comment by nehadnb** [comment]
@rijkvanzanten Do you have some samples that says repository should have this files and content will be this, What env variable add in digital ocean? 

I am new to directus and digitalocean, Can you please help me out?

**Reply by paescuj** [reply]
Just a friendly side note: As you can imagine the team members have already tons of other work to do, so pinging them is not really conducive and is also not very welcome in my opinion.

**Comment by paescuj** [comment]
Most likely you're missing some environment variables like `KEY` and `SECRET`, see https://docs.directus.io/self-hosted/config-options.html#security.

Note that due to #16284 not all log entries are getting printed to the console, so you might want to define the environment variable `LOG_STYLE=raw` as a workaround until the next release. Once defined you will be informed about missing environment variables:
```console
{"level":50,"time":1671312187042,"pid":86179,"hostname":"<...>","msg":"\"KEY\" Environment Variable is missing."}
```

**Reply by nehadnb** [reply]
After adding KEY and SECRET and LOG_STYLE it is same issue
![image](https://user-images.githubusercontent.com/109754993/208523851-6b4d53a2-88d3-41ac-a36c-2bb5b212ebea.png)

**Comment by rijkvanzanten** [comment]
I would highly recommend using the docker container instead of the direct Node setup 👍🏻

**Comment by nehadnb** [comment]
Is there any documantaion that provide how to deploy directus with docker in digitalocean?

**Reply by paescuj** [reply]
I've no experience with DigitalOcean but I guess it's pretty straightforward:

1. Create a new App
2. Use the image from Docker Hub:
    <img width="500" src="https://user-images.githubusercontent.com/5363448/208501385-e28af420-c218-4411-91a4-85d39f2f0472.png">
3. Probably add a database resource:
    <img width="500" src="https://user-images.githubusercontent.com/5363448/208501631-b4cf8f9b-efb3-45d4-b556-5b4f0e113dc3.png">
4. I guess that's the name under which the DB will be available inside the Directus container:
    <img width="500" src="https://user-images.githubusercontent.com/5363448/208502243-230765bc-b688-49b3-84eb-1e66e82c0658.png">
5. Define the environment variables (as you already did before):
    <img width="500" src="https://user-images.githubusercontent.com/5363448/208501772-c16c6a6b-f9b3-412a-a25c-1a4e91317aaf.png">
6. Ready to go!

**Reply by nehadnb** [reply]
Thanks for this, using this i can not manage custom changes repository

**Reply by paescuj** [reply]
> Thanks for this, using this i can not manage custom changes repository

What do you mean by this? Do you want to use custom extensions or run a modified version of the core itself?

**Reply by nehadnb** [reply]
I mean if i want to create new layout or module and keep track of that changes so i am using gitlab/github

**Reply by paescuj** [reply]
Okay, so instead of the "Docker Hub" source you should be able to use a repository. There you can store the source of your extensions and a custom `Dockerfile` which is based on `directus/directus` and does build and include the extensions.

**Reply by nehadnb** [reply]
Right, i have repository and using it and same thing works properly in local but not in digitalocean because if i add npx directus bootstrap it is giving error 
![image](https://user-images.githubusercontent.com/109754993/208718288-afac3440-aa23-4362-bf48-6a13509db47b.png)

is there any other way to run "npx directus bootstrap"? How can i solve that? 

Without that command it will fail in deploy mode and not showing any error

**Reply by paescuj** [reply]
Sorry for my late reply here! My idea was to switch completely to Docker instead of using the Node.js package. This would mean you won't have to call `npx directus bootstrap` at all.

**Comment by nehadnb** [comment]
it is working after move "npx directus bootstrap" command from Build Command to Run Command

**Reply by paescuj** [reply]
If you're sticking to the "Node.js way" this makes very much sense! 👍

**Reply by rick1290** [reply]
Can you post the full package file for others to see how you set it uup?
